src/arr/trove/ast.arr

```diff
@@ -490,6 +525,34 @@ sharing:
     self._match(visitor, lam(val): raise("No visitor field for " + self.label()) end)
   end
 end
+data DefinedModule:
+  | s-defined-module(name :: String, provided-name :: Option<Name>, id :: Name, mod :: String) with:
```

Comment:
Is `mod` intended to be a URI?

Reply:
In this and the preceding examples, it's not at all clear what the difference between `name` and `provided-name` are. Some description/example would be helpful near these types.